room_alignment/lifecycle.py

```
@@ -0,0 +1,118 @@
+    try:
+        fcntl.flock(lock_file.fileno(), fcntl.LOCK_EX | fcntl.LOCK_NB)
+    except BlockingIOError:
+        return False
+    return True
+
+
+def stop(data_dir: Path, timeout_seconds: float = 10.0, force: bool = False) -> dict[str, object]:
+    """Stop the process owning one Room Alignment state directory."""
+
+    if timeout_seconds < 0:
+        raise ValueError("Stop timeout must be zero or greater")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)